File: frontend/lib/model/order_item.dart
class OrderItem {
  final String pizzaId, pizzaName, imageUrl;
  final int quantity;

  OrderItem({
    required this.pizzaId,
    required this.pizzaName,
    required this.imageUrl,
    required this.quantity,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      pizzaId: json['pizzaId'] as String,
      pizzaName: json['pizzaName'] as String,
      imageUrl: json['imageUrl'] as String,
      quantity: (json['quantity'] as num).toInt(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'pizzaId': pizzaId,
      'pizzaName': pizzaName,
      'imageUrl': imageUrl,
      'quantity': quantity,
    };
  }
}

File: dashboard/src/model/order_item.ts
export interface RawOrderItem {
  pizzaId: string;
  pizzaName: string;
  imageUrl: string;
  quantity: number;
}

export interface OrderItem {
  pizzaName: string;
  quantity: number;
}

export function createOrderItems(rawItems: RawOrderItem[]): OrderItem[] {
  return rawItems.map((item) => ({
    pizzaName: item.pizzaName,
    quantity: item.quantity,
  }));
}